Authorization/Authorization/Presentation/Base/FieldsView.swift

```diff
@@ -54,23 +56,67 @@ struct FieldsView: View {
             case .checkbox:
                 EmptyView()
                     .id(index)
-                Text("Checkbox is not support")
             case .plaintext:
-                HTMLFormattedText(
-                    cssInjector.injectCSS(
-                        colorScheme: colorScheme,
-                        html: config.field.label,
-                        type: .discovery,
-                        fontSize: 90, screenWidth: proxy.size.width)
-                )
-                .id(UUID())
-                .padding(.horizontal, -6)
-                
+                plaintext(config: config)
             case .unknown:
                 Text("This field not support")
             }
         }
     }
+
+    @ViewBuilder
+    private func plaintext(config: FieldConfiguration) -> some View {
+        if config.field.isHonorCode,
+           let eulaURL = self.config.agreement.eulaURL,
+           let tosURL =  self.config.agreement.tosURL,
+           let policy = self.config.agreement.privacyPolicyURL {
+            let text = AuthLocalization.SignUp.agreement(
+                "\(self.config.platformName)",
+                eulaURL,
+                "\(self.config.platformName)",
+                tosURL,
+                policy
+            )
+            let checkBox = fields.first(where: { $0.field.type == .checkbox })
+            checkBox.flatMap { _ in
+                CheckBoxView(
+                    checked: $sendMarketing,
+                    text: AuthLocalization.SignUp.marketingEmailTitle("\(self.config.platformName)"),
+                    font: Theme.Fonts.labelSmall
+                )
+                .padding(.vertical, 10)
+                .onAppear {
+                    checkBox?.text = "\(sendMarketing)"
+                }
+                .onChange(of: sendMarketing) { newValue in
+                    checkBox?.text = "\(newValue)"
```

Comment:
Any reason to assign boolean text to text of the checkbox? I've commented the code and it was working fine.